sql/sql_base.cc

```diff
@@ -2245,6 +2245,43 @@ static TABLE_LIST* find_dup_table(THD *thd, const TABLE_LIST *table,
   DBUG_RETURN(res);
 }
 
+/**
+  Test that the subject table (of DELETE) has no cascade foreign key
+  parent present in the query.
+
+  @param  thd        thread handle
+  @param  table      table to be checked (must be updatable base table)
+  @param  table_list List of tables to check against
+
+  @retval non-NULL The table list element for the table that
+                   represents the cascade parent.
+  @retval NULL     No cascade parents found.
+*/
+TABLE_LIST *no_cascade_dependency(THD *thd, const TABLE_LIST *table,
+                                  TABLE_LIST *table_list)
+{
+  DBUG_ASSERT(table == ((TABLE_LIST *)table)->updatable_base_table());
+
+  List <st_handler_tablename> fk_table_list;
+  List_iterator<st_handler_tablename> fk_table_list_it(fk_table_list);
+  st_handler_tablename *tbl_name;
+
+  table->table->file->get_cascade_foreign_key_table_list(thd, &fk_table_list);
+
+  while ((tbl_name= fk_table_list_it++))
+  {
+    for (TABLE_LIST* curr = table_list; curr; curr= curr->next_local)
+    {
+      if (strcmp(curr->table_name, tbl_name->tablename) == 0 &&
```

Comment:
This is not going to work with different lowercase_table_name settings - check for how that variable is used for canonical way to compare names which takes case insensitivity into account